library/src/main/java/com/google/android/exoplayer2/extractor/flv/VideoTagPayloadReader.java

```diff
@@ -92,7 +93,9 @@ protected void parsePayload(ParsableByteArray data, long timeUs) throws ParserEx
           Format.NO_VALUE, Format.NO_VALUE, avcConfig.width, avcConfig.height, Format.NO_VALUE,
           avcConfig.initializationData, Format.NO_VALUE, avcConfig.pixelWidthAspectRatio, null);
       output.format(format);
-      hasOutputFormat = true;
+      if (listener != null) {
```

Comment:
I guess we don't receive the packet types very often. I only see this packet received when the player is initialized and the configuration of the stream is changed. I added the format change check in the implementation of the listener because I guess for me it might be a little easier to debug XD